src/server/game/Entities/Object/Object.cpp

```diff
@@ -2558,7 +2558,7 @@ float WorldObject::GetFloorZ() const
 {
     if (!IsInWorld())
         return m_staticFloorZ;
-    return std::max<float>(m_staticFloorZ, GetMap()->GetGameObjectFloor(GetPhaseMask(), GetPositionX(), GetPositionY(), GetPositionZ()));
+    return std::max<float>(m_staticFloorZ, GetMap()->GetHeight(GetPositionX(), GetPositionY(), GetPositionZ()));
```

Comment:
Why? m_staticFloorZ should be exactly equal to Map::GetHeight (the call without checking _dynamicTree)